tests/test_core.py

```diff
@@ -891,6 +891,24 @@ def test_shape_only_ops(fn: Callable, tmp_path_factory: pytest.TempPathFactory)
         result.compute()
 
 
+def test_assign_behavior() -> None:
+    x = ak.Array([{"a": 1, "b": 2}, {"a": 3, "b": 4}], behavior={}, attrs={})
+    dx = dak.from_awkward(x, 3)
```

Comment:
Should test that `dx`'s behavior and attrs are the ones given (would have been `None`)

Reply:
See other changes. Unless we don't want to take the array's original attrs if none given.

Reply:
OK this exposed a few places where attrs weren't being propagated correctly (in particular the whole from-json interface). That is fixed now.